game-core/src/main/java/games/strategy/triplea/ui/MapPanel.java

```diff
@@ -458,11 +458,10 @@ public void gameDataChanged(final Change change) {
       if (!playersWithTechChange.isEmpty()
           || UnitIconProperties.getInstance(gameData).testIfConditionsHaveChanged(gameData)) {
         tileManager.resetTiles(gameData, uiContext.getMapData());
-        try {
-          SwingUtilities.invokeAndWait(() -> repaint());
-        } catch (final Exception e) {
-          // ignore failed repaint
-        }
+        SwingUtilities.invokeLater(() -> {
+          recreateTiles(getData(), uiContext);
```

Comment:
Ugghh I just noticed recreateTiles is a very time sensitive call that should really not be called on the EDT.
On a slow machine this will freeze the display for a noticeable amount of time

Reply:
I noticed this as well though at least before all the other tile changes, this seemed to be the only way to force them to redraw. Probably can try removing this now to see if just the repaint call works now.